sw/device/lib/dif/dif_sram_ctrl.h

```diff
@@ -0,0 +1,236 @@
+
+/**
+ * The result of an attempt to enable or disable execution from SRAM.
+ */
+typedef enum dif_sram_ctrl_exec_result {
+  /**
+   * Indicates that the operation succeeded.
+   */
+  kDifSramCtrlExecOk = kDifSramCtrlOk,
+  /**
+   * Indicates some unspecified failure.
+   */
+  kDifSramCtrlExecError = kDifSramCtrlError,
+  /**
+   * Indicates that some parameter passed into a function failed a
+   * precondition.
+   *
+   * When this value is returned, no hardware operations occured.
+   */
+  kDifSramCtrlExecBadArg = kDifSramCtrlBadArg,
+  /**
+   * Indicates that this operation has been locked out, and can never
+   * succeed until hardware reset.
+   */
+  kDifSramCtrlExecLocked,
+} dif_sram_ctrl_exec_result_t;
+
+/**
+ * Enable or disable execution from SRAM.
+ *
+ * @param sram_ctrl A SRAM Controller handle.
+ * @param state Enable or disable SRAM execution.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_sram_ctrl_exec_result_t dif_sram_ctrl_exec_set_enabled(
+    const dif_sram_ctrl_t *sram_ctrl, const dif_sram_ctrl_toggle_t *state);
+
+/**
+ * Report whether execution from SRAM is currently enabled or disabled.
+ *
+ * @param sram_ctrl A SRAM Controller handle.
+ * @param[out] state State of SRAM execution (enabled or disabled).
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_sram_ctrl_result_t dif_sram_ctrl_exec_get_enabled(
+    const dif_sram_ctrl_t *sram_ctrl, dif_sram_ctrl_toggle_t *state);
+
+/**
+ * Retrieve the error state from the SRAM controller.
+ *
+ * This function will return `kDifSramCtrlOk` if the error state of the
+ * SRAM controller was successfully fetched regardless of whether the SRAM
+ * controller has experienced an uncorrectable parity error or not. The `error`
+ * argument must be checked to see if a uncorrectable parity error has actually
+ * occurred.
+ *
+ * @param sram_ctrl A SRAM Controller handle.
+ * @param[out] error Whether an uncorrectable parity error has occurred.
+ * @param[out] address The byte address of the last parity error (optional).
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_sram_ctrl_result_t dif_sram_ctrl_get_error(const dif_sram_ctrl_t *sram_ctrl,
+                                               bool *error, uintptr_t *address);
+
+/**
+ * Retrieve the current key validity from the SRAM controller.
+ *
+ * If `valid` is set to true then SRAM is scrambled using a valid key from OTP.
+ * Otherwise SRAM is scrambled using a default all-zero key and nonce.
+ *
+ * @param sram_ctrl A SRAM Controller handle.
+ * @param[out] valid Whether a key has been successfully obtained.
+ * @param[out] seed_valid Whether the scrambling key was derived from a valid
+ * key seed (optional).
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_sram_ctrl_result_t dif_sram_ctrl_get_key_valid(
+    const dif_sram_ctrl_t *sram_ctrl, bool *valid, bool *seed_valid);
```

Comment:
I don't like two bool params that could accidentally get transposed. Why not a struct?

Reply:
I agree. Converted into a struct.